### PR TITLE
UI-38 Build RadioGroup component

### PR DESCRIPTION
**Pull request by PiotrGrzechnik**
#### Task

- **Issue or task referred:** UI-38
- **Feature or Bugfix:** Feature

Build component which represents radio group. It's a wrapper for RadioInput components which accepts array of objects with value and label for each option. 

#### Checklist

- [x] Required unit tests prepared
- [x] Documentation prepared
- [x] External licenses validated (ideally MIT)

### CODE REVIEW

File: packages/radio-group/src/RadioGroup.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @param {string|number} [props.value]
+ * @param {array} [props.disabled]
+ * @returns {React.Element}
+ */
+function RadioGroup (props) {
+  const { className, children, name, options, value, size, disabled, ...passedProps } = props
+  return (
+    <div className={buildClassName('radio-group', className)} {...passedProps} >
+      {options.map((obj) =>
+        <RadioInput key={obj.value} name={name} value={obj.value} size={size} defaultChecked={value === obj.value} disabled={disabled.includes(obj.value)} {...passedProps} >
```
Comment (michalochman) [inline packages/radio-group/src/RadioGroup.js]: 1) Please split props into separate lines.
2) I would put `label` into `key`.
3) It looks like `RadioGroup` is controlled by `value`, in that case we should use `checked` instead of `defaultChecked`.
4) Unnecessary whitespace between `{...passedProps}` and `>`.
5) @rangoo94 disabled prop seems awkward to me, maybe it should be a member of option?

File: packages/radio-group/src/RadioGroup.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+RadioGroup.propTypes = {
+  /** Additional class name */
+  className: PropTypes.string,
+
+  /** Name of radio group */
+  name: PropTypes.string.isRequired,
+
+  /** Array of objects which represent options */
+  options: PropTypes.array.isRequired,
```
Comment (rangoo94) [inline packages/radio-group/src/RadioGroup.js]:
```js
PropTypes.arrayOf(PropTypes.shape({
  label: PropTypes.node.isRequired,
  value: PropTypes.any.isRequired,
  disabled: PropTypes.bool
}))
```

File: packages/radio-group/stories.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+// Create factories for story
+const addStory = createStoriesFactory('RadioGroup', module, {
+  propTables: [ RadioGroup ]
+})
+
+// Stories
+
+addStory.controlled('initial', readme, (setState, state) => (
+  <RadioGroup onChange={value => setState({ value })} name='RadioName' size='small' value={state.value} options={[{value: 1, label: 'one'}, {value: 2, label: 'two'}, {value: 3, label: 'three'}]} />
```
Comment (rangoo94) [inline packages/radio-group/stories.js]: Could you put it to new lines? It's waaaay too big.

```
<RadioGroup
  onChange={value => setState({ value })}
  name='RadioName'
  size='small'
  value={state.value}
  options={[{value: 1, label: 'one'}, {value: 2, label: 'two'}, {value: 3, label: 'three'}]}
/>
```

Also, why `initial` story has small size? shouldn't it be without that?

File: packages/radio-group/src/RadioGroup.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      {options.map((obj) =>
+        <RadioInput
+          checked={value === obj.value}
+          disabled={obj.disabled || false}
+          key={obj.label}
+          name={name}
+          onChange={checked => checked && onChange && onChange(obj.value)}
+          size={size}
+          value={obj.value}
+          {...passedProps}>
```
Comment (rangoo94) [inline packages/radio-group/src/RadioGroup.js]: Do not pass all props to all `input`s please. Just to `radio-group`.

File: packages/radio-group/styles/main.sass [hunk 5]
```diff
@@ -0,0 +1,2 @@
+@import './geometry'
```
Comment (rangoo94) [inline packages/radio-group/styles/main.sass]: Remove `storybook.sass`.

File: packages/radio-group/tests/RadioGroup.test.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+
+import RadioGroup from '../src/RadioGroup'
+
+describe('<RadioGroup />', () => {
+  it('renders children correctly', () => {
+    const wrapper = shallow(<RadioGroup name='RadioGroup' options={[{value: 1, label: 'one'}, {value: 2, label: 'two'}, {value: 3, label: 'three'}]} />)
```
Comment (rangoo94) [inline packages/radio-group/tests/RadioGroup.test.js]: Could you split it to many lines?

Comment (rangoo94) [reply packages/radio-group/tests/RadioGroup.test.js]: Just to note: most important rule is to keep lines smaller than display size - right now I have to scroll to see the rest of `RadioGroup`

File: packages/radio-input/src/RadioInput.js [hunk 7]
```diff
@@ -14,13 +14,13 @@ import { buildClassName } from '@talixo/shared'
...
 function RadioInput (props) {
-  const { children, className, size, style, ...passedProps } = props
+  const { children, className, size, style, onChange, ...passedProps } = props
 
   const wrapperClass = buildClassName('radio-input', className, [ size ])
 
   return (
     <label className={wrapperClass} style={style}>
-      <input type='radio' {...passedProps} />
+      <input type='radio' onChange={e => onChange && onChange(e.target.checked)} {...passedProps} />
```
Comment (rangoo94) [inline packages/radio-input/src/RadioInput.js]: Could you extract it above?

File: packages/radio-group/src/RadioGroup.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+ * @param {array} [props.options]
+ * @param {string} [props.size]
+ * @param {string|number} [props.value]
+ * @returns {React.Element}
+ */
+function RadioGroup (props) {
+  const { className, children, name, options, value, size, onChange, ...passedProps } = props
+  return (
+    <div className={buildClassName('radio-group', className)} {...passedProps} >
+      {options.map((obj) =>
```
Comment (rangoo94) [inline packages/radio-group/src/RadioGroup.js]: Could you extract it above and wrap component using `()`?

File: packages/radio-group/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+- `prop-types: ^15.6.1`
+- `react: ^16.2.0`
+- `react-dom: ^16.2.0`
+
+These packages are required by `@talixo/radio-group`, but you have to install them manually,
+to avoid having different versions of these in your application.
+
+## Supported props
+
```
Comment (maciek15) [inline packages/radio-group/README.md]: It also passes all additional props to `div`

**Comment by PiotrGrzechnik** [comment]
@stuniel It's not necessary in RadioGroup

File: packages/radio-group/src/RadioGroup.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    label: PropTypes.node.isRequired,
+    value: PropTypes.any.isRequired,
+    disabled: PropTypes.bool
+  })),
+
+  /** Radio input label size ('small', 'large') */
+  size: PropTypes.oneOf([ 'small', 'large' ]),
+
+  /** Value of default option */
+  value: PropTypes.oneOfType([
```
Comment (rangoo94) [inline packages/radio-group/src/RadioGroup.js]: When option `value` is set to `PropTypes.any`, then base `value` should be `any` too

File: packages/radio-group/package.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@talixo/radio-group",
+  "version": "0.1.0",
```
Comment (rangoo94) [inline packages/radio-group/package.json]: `0.0.0`

File: packages/radio-group/package.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    "type": "git",
+    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
+  },
+  "bugs": {
+    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
+  },
+  "license": "MIT",
+  "devDependencies": {
+    "@talixo/shared": "^0.1.0",
+    "@talixo/radio-input": "^0.1.0",
```
Comment (rangoo94) [inline packages/radio-group/package.json]: Radio input should be in `peerDependencies` as well - you require it.
